WebRoot/public/login.jsp

```diff
@@ -384,6 +385,29 @@ if (application.getInitParameter("offlineMode") != null) {
 </c:url>
 
 <%
+	Cookie testCookieRootPath = new Cookie("ZM_TEST", "true");
+	testCookieRootPath.setMaxAge(0);
+	testCookieRootPath.setPath("/");
+	response.addCookie(testCookieRootPath);
```

Comment:
this is already done at line no 413, are we repeating it?

Reply:
Before the fix, for example,
1. access `https://host`. A cookie for `/` is created.
2. access `https://host/test/path`. A cookie for `/test/path` is created.
3. access `https://host`. The cookie for `/` is updated. The cookie for `/test/path` still remains and the value is not updated. 
4. access `https://host/test/path`. The cookie for `/test/path` is update. The cookie for `/` still remains and the value is not updated. 

The unnecessary cookie whose path is different from the access path needs to be removed.
On JSP side, `request.getContext` or `getRequestURI` always returns `/test/path`. Then the jsp cannot identify the access URL specified on the browser.
So the fix is to remove both cookies first before line 413.

Reply:
not sure why we are rendering login page when accessing https://host and zimbraMailURL is set, nginx should send a redirect to correct address, this way we don't have to handle these scenarios in login.jsp

Reply:
I get your point. I will update it.

Reply:
Updated the PR.
The issue is solved by specify the path `/` for ZM_LOGIN_CSRF and ZM_TEST cookies. The cookies will work on both `https://host/` and `https://host/test/path`.